src/compas_viewer/scene/sceneobject.py

```diff
@@ -287,6 +287,7 @@ def init(self):
         self.make_buffers()
         self._update_matrix()
         self.instance_color = Color.from_rgb255(*next(self.viewer.scene._instance_colors_generator))
+        # TODO: check robot add instance color error
         self.scene.instance_colors[self.instance_color.rgb255] = self
```

Comment:
But what I can do here is to change to `self.viewer.scene` just like L289 so it will be able to access it no matter when, I will push it to main now

Reply:
Thats very cool. Then should we just use `self.viewer.scene` instead of `self.scene` in other place as well? cus it access the same instance anyway?